src/media_tools/av_parsers.c

```diff
@@ -1545,6 +1545,595 @@ GF_Err gf_m4a_write_config(GF_M4ADecSpecInfo *cfg, char **dsi, u32 *dsi_size)
+		}
+		GF_LOG(GF_LOG_DEBUG, GF_LOG_CONTAINER, ("[AV1] Annex B    frame unit detected (size "LLU")\n", frame_unit_size));
+		sz -= Leb128Bytes + frame_unit_size;
+
+		while (frame_unit_size > 0) {
+			ObuType obu_type;
+			u64 obu_length = leb128(bs, &Leb128Bytes);
+			if (frame_unit_size < Leb128Bytes + obu_length) {
+				GF_LOG(GF_LOG_ERROR, GF_LOG_CODING, ("[AV1] Annex B frame_unit_size("LLU") < Leb128Bytes("LLU") + obu_length("LLU")\n", frame_unit_size, Leb128Bytes, obu_length));
+				return GF_NON_COMPLIANT_BITSTREAM;
+			}
+			GF_LOG(GF_LOG_DEBUG, GF_LOG_CONTAINER, ("[AV1] Annex B        OBU detected (size "LLU")\n", obu_length));
+			u64 pos = gf_bs_get_position(bs);
+			frame_unit_size -= Leb128Bytes;
+
+			e = gf_media_aom_av1_parse_obu(bs, &obu_type, &obu_length, state);
+			if (e) return e;
+			assert(obu_length == gf_bs_get_position(bs) - pos);
+
+			av1_populate_state_from_obu(bs, pos, obu_length, obu_type, state);
+			frame_unit_size -= obu_length;
+		}
+
+		assert(frame_unit_size == 0);
+	}
+
+	assert(sz == 0);
+
+	return GF_OK;
+}
+
+GF_Err aom_av1_parse_obu_from_ivf(GF_BitStream *bs, AV1State *state)
+{
+	u64 frame_size;
+	GF_Err e = gf_media_aom_parse_ivf_frame_header(bs, &frame_size);
+	if (e) return e;
+	GF_LOG(GF_LOG_DEBUG, GF_LOG_CONTAINER, ("[AV1] IVF frame detected (size "LLU")\n", frame_size));
+
+	while (frame_size > 8) {
+		u64 obu_size = 0, pos = gf_bs_get_position(bs);
+		ObuType obu_type;
+
+		if (gf_media_aom_av1_parse_obu(bs, &obu_type, &obu_size, state) != GF_OK)
+			return e;
+		assert(obu_size == gf_bs_get_position(bs) - pos);
+
+		av1_populate_state_from_obu(bs, pos, obu_size, obu_type, state);
+
+		frame_size -= obu_size;
+	}
+
+	return e;
+}
+
+typedef enum {
+	KEY_FRAME = 0,
+	INTER_FRAME = 1,
+	INTRA_ONLY_FRAME = 2,
+	SWITCH_FRAME = 3,
+} AV1FrameType;
+
+static void av1_parse_uncompressed_header(GF_BitStream *bs, AV1State *state) {
+	AV1StateFrame *frame_state = &state->frame_state;
+	if (state->frame_id_numbers_present_flag) {
+		//idLen = (additional_frame_id_length_minus_1 + delta_frame_id_length_minus_2 + 3);
+	}
+	//allFrames = (1 << NUM_REF_FRAMES) - 1;
+	if (state->reduced_still_picture_header) {
+		frame_state->key_frame = GF_TRUE;
+	} else {
+		Bool show_existing_frame = gf_bs_read_int(bs, 1);
+		if (show_existing_frame == GF_TRUE) {
+			/*frame_to_show_map_idx	f(3)
+			if (decoder_model_info_present_flag && !equal_picture_interval) {
+				temporal_point_info()
+			}
+			refresh_frame_flags = 0
+			if (frame_id_numbers_present_flag) {
+				display_frame_id	f(idLen)
+			}
+			frame_type = RefFrameType[frame_to_show_map_idx]
+			if (frame_type == KEY_FRAME) {
+				refresh_frame_flags = allFrames
+			}
+			if (film_grain_params_present) {
+				load_grain_params(frame_to_show_map_idx)
+			}*/
+			return;
+		}
+		AV1FrameType frame_type = gf_bs_read_int(bs, 2);
+		frame_state->key_frame = gf_bs_read_int(bs, 1)/*show_frame*/ && frame_type == KEY_FRAME && frame_state->seen_frame_header;
```

Comment:
Could we split that on two lines for readability: reading the show_frame flag and then setting the key frame state? 
Also,  you set key_frame to true without checking the presence of sequence header in the same temporal unit?

Reply:
> Could we split that on two lines for readability: reading the show_frame flag and then setting the key frame state?

Done.

> Also, you set key_frame to true without checking the presence of sequence header in the same temporal unit?

Correct. I skipped that check because the AV1 doesn't seem to prevent having the sequence header either 1) several times or 2) after the frame header. In this implementation we naively overwrite in the latest parsed struct.

I re-read the AV1 in ISOBMF document (```It contains a Sequence Header before the first Frame Header.```) and I think I was too conservative. So I've added the check.